api/v1alpha1/releaseplan_webhook_test.go

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-appstudio/release-service/kcp"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//+kubebuilder:scaffold:imports
+)
+
+var _ = Describe("ReleasePlan webhook", func() {
+	var releasePlan *ReleasePlan
+
+	BeforeEach(func() {
+		releasePlan = &ReleasePlan{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "appstudio.redhat.com/v1alpha1",
+				Kind:       "ReleasePlan",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "releaseplan",
+				Namespace: "default",
+			},
+			Spec: ReleasePlanSpec{
+				DisplayName: "Test release plan",
+				Application: "application",
+				Target: kcp.NamespaceReference{
+					Namespace: "default",
+				},
+			},
+		}
+	})
+
+	AfterEach(func() {
+		err := k8sClient.Delete(ctx, releasePlan)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+	})
+
+	Context("When a ReleasePlan is created without the auto-release label", func() {
+		It("should get the label added with its value set to true", func() {
+			Expect(k8sClient.Create(ctx, releasePlan)).Should(Succeed())
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{
+					Name:      releasePlan.Name,
+					Namespace: releasePlan.Namespace,
+				}, releasePlan)
+
+				labelValue, ok := releasePlan.GetLabels()[AutoReleaseLabel]
+
+				return err == nil && ok && labelValue == "true"
+			}, timeout).Should(BeTrue())
+		})
+	})
+
+	Context("When a ReleasePlan is created with an invalid auto-release label value", func() {
+		It("should get rejected until the value is valid", func() {
+			releasePlan.Labels = map[string]string{AutoReleaseLabel: "foo"}
+			err := k8sClient.Create(ctx, releasePlan)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("'%s' label can only be set to true or false", AutoReleaseLabel))
```

Comment:
Also says _until_. Sounds like an `Eventually` block is required.

Reply:
Same as before. Manual user action.